### ISSUE TITLE
Possible to hide \unicharchart header row?

### ISSUE THREAD

**Issue by markpenny** [silnrsi/fontproof#11]
I really like this tool and managed to produce something pretty decent after learning the ropes. But I have a request for the \unicharchart:
If the top "header" row contains no variation at all, then it should be suppressed.
So here, it is welcome, and useful:
![image](https://user-images.githubusercontent.com/12962708/73347595-e0696380-42ad-11ea-9545-79320bc779f4.png)

But here it isn't useful or needed at all.
![image](https://user-images.githubusercontent.com/12962708/73347612-e6f7db00-42ad-11ea-84a1-513da3bf08a0.png)

If it doesn't automatically suppress itself, then could there be a parameter to manually choose when to suppress it? Thanks for considering this improvement.

**Comment by devosb** [comment]
Commit https://github.com/silnrsi/fontproof/commit/d7af29d41a3c227f187e3b693a21a71bb7f4131c fixes the header to have four hex digits (instead of three) for characters in the SMP, as shown below
![unicharchart](https://user-images.githubusercontent.com/12450072/73360751-dec57e80-4260-11ea-89e4-3509b6819c0f.png)
Unfortunately, this fix has not made it into package yet, which is why you did not see the fix. Hopefully an updated package will be made in a few months.

With this fix, is a suppress option still needed?

**Comment by alerque** [comment]
You could easily implement this yourself. If you want to trigger the change from the CLI like this project does for fonts, just apply a patch like this:

```patch
diff --git a/classes/fontproof.lua b/classes/fontproof.lua
index 4d95ea0..53c29c5 100644
--- a/classes/fontproof.lua
+++ b/classes/fontproof.lua
@@ -405,7 +405,8 @@ SILE.registerCommand("unicharchart", function (options, _)
   local done = 0
   while done < #glyphs do
     -- header row
-    if type == "range" then
+    -- luacheck: ignore disableheader
+    if type == "range" and not disableheader then
       SILE.call("font", { size = charsize }, function()
         for j = 0, columns-1 do
           local ix = done + j * rows
```

Then run it with the disable flag set:

    sile -e 'disableheader = true' test.sil

You could also do this by passing an option to the unicharchart command:

```patch
diff --git a/classes/fontproof.lua b/classes/fontproof.lua
index 4d95ea0..31e8fb7 100644
--- a/classes/fontproof.lua
+++ b/classes/fontproof.lua
@@ -367,6 +367,7 @@ end)
 
 SILE.registerCommand("unicharchart", function (options, _)
   local type = options.type or "all"
+  local showheader = SU.boolean(options.showheader, true)
   local rows = tonumber(options.rows) or 16
   local columns = tonumber(options.columns) or 12
   local charsize = tonumber(options.charsize) or 14
@@ -405,7 +406,7 @@ SILE.registerCommand("unicharchart", function (options, _)
   local done = 0
   while done < #glyphs do
     -- header row
-    if type == "range" then
+    if type == "range" and showheader then
       SILE.call("font", { size = charsize }, function()
         for j = 0, columns-1 do
           local ix = done + j * rows
```

Then invoke like so:

```tex
\unicharchart[showheader=false]
```

Or from Lua:

```lua
SILE.call("unicharchart", { showheader = false })
```

I'd be happy to contribute a PR that does either of these, but I'm not sure which use case is more appropriate for the way the project actually gets used.

**Comment by alerque** [comment]
@devosb That commit was included in master and has been in release tags v1.2.0, v1.3.0, and v1.3.1.

**Comment by devosb** [comment]
Indeed, but @markpenny is using the sile package (and the package bundles FontProof as well) from https://launchpad.net/~silnrsi/+archive/ubuntu/smith-py3 That is the package I was referring to.

As for the suppress header option, the second use case would be better. @markpenny sent me an email with an example where the fix I coded would help with one chart, but has a second chart where my fix does not help, and a suppress header option would help.

I will let @jvgaultney handle any PRs. Thank you for your help.

**Comment by alerque** [comment]
@devosb Are you saying this is somehow bundled inside the tar being released in that PPA as "sile-0.9.5.1", but is really a modified version with extra stuff hacked in? That's ... um ... dirty¹. I would really like to get a proper Ubuntu package released for SILE (see [this upstream issue](https://github.com/sile-typesetter/sile/issues/638)) that is clean and up to date. Maybe fontproof can be packaged separately in that PPA so it doesn't conflict with other installs of SILE (whether from source or some other packaging).

¹ "Dirty" in the technical sense, see for example `man git describe` and its concept of a dirty working tree.

**Comment by n7s** [comment]
Notice the actual -sile tag to the package, yes, this was our quick and dirty solution at the time, separate packages for extensions are in the works. I don't think this has negatively impacted any sile users out there that our PPA had a repackaged tarball. Rest assured I did not intentionnally change any other source items.
